File: tests_deps/src/main/java/com/github/klee0kai/test/boxed/model/CarBoxedInject.java
package com.github.klee0kai.test.boxed.model;

import com.github.klee0kai.test.car.model.Bumper;
import com.github.klee0kai.test.car.model.Wheel;
import com.github.klee0kai.test.car.model.Window;

import javax.inject.Inject;

public class CarBoxedInject {


    @Inject
    public CarBox<Bumper> bumper;

    @Inject
    public CarBox<Wheel> wheel;

    @Inject
    public CarBox<Window> window;

    public CarBox<Bumper> bumperFromMethod;

    public CarBox<Wheel> wheelFromMethod;

    public CarBox<Window> windowFromMethod;


    @Inject
    public void init(CarBox<Bumper> bumper, CarBox<Wheel> wheel, CarBox<Window> window) {
        bumperFromMethod = bumper;
        wheelFromMethod = wheel;
        windowFromMethod = window;
    }

}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/boxed/model/CarBoxedInject.kt
package com.github.klee0kai.test.boxed.model

import com.github.klee0kai.test.car.model.Bumper
import com.github.klee0kai.test.car.model.Wheel
import com.github.klee0kai.test.car.model.Window
import javax.inject.Inject

class CarBoxedInject {
    @Inject
    lateinit var bumper: CarBox<Bumper>

    @Inject
    lateinit var wheel: CarBox<Wheel>

    @Inject
    lateinit var window: CarBox<Window>


    var bumperFromMethod: CarBox<Bumper>? = null
    var wheelFromMethod: CarBox<Wheel>? = null
    var windowFromMethod: CarBox<Window>? = null

    @Inject
    fun init(bumper: CarBox<Bumper>?, wheel: CarBox<Wheel>?, window: CarBox<Window>?) {
        bumperFromMethod = bumper
        wheelFromMethod = wheel
        windowFromMethod = window
    }
}
